### ISSUE TITLE
impress endless scrolling

### ISSUE THREAD

**Issue by Andreas-Kainz** [CollaboraOnline/online#5118]
I understood that on desktop apps like in LibreOffice Impress you have the sidebar with the previews and in the main section the actual slide.

On mobile it's way easier to slide through a presentation (like you do in writer where you slide from one page to the next one) than you click one the previews. I understood that it's easier for **editing** one page after another and you see only one page, but on mobile you ordinary view content and for viewing it's way easier to scroll through the impress slide like you do at writer text files.

#### Before
![Screenshot_20220807_174013](https://user-images.githubusercontent.com/8517736/183299115-a9a64373-2a37-419d-a48a-0812aeec288b.png)

#### Proposal
![new](https://user-images.githubusercontent.com/8517736/183299107-a7acdfae-45eb-4d1c-9dd4-96354d423871.png)

**Comment by mikekaganski** [comment]
Note https://bugs.documentfoundation.org/show_bug.cgi?id=38164, implemented for core 24.8. While not showing several slides at once (which I believe is not easy and not really needed), the switching between slides by sliding below/above (as done in desktop now) should be reasonably straightforward to implement.

**Comment by mmeeks** [comment]
@pedropintosilva AFAIR we do this already for PDF, were there reasons around editing shapes not to do this somehow by default (?) - perhaps @gokaysatir has a view.

**Comment by pedropintosilva** [comment]
@mohit-marathe whenever you have time investigate this but remember to sync with @quikee (WRT his rendering experimentation) and @gokaysatir .

**Comment by mohit-marathe** [comment]
@pedropintosilva https://gerrit.collaboraoffice.com/c/online/+/1396 enables file based view (Writer like scrolling) in Impress when in `Viewing mode`.

**Comment by gokaysatir** [comment]
@mmeeks yes this goal was a bit more difficult before. Because of the initial work it required. And we had leaflet. Now we are using our own implementation for document view. It is easier now to accomplish this task I hope.

@mohit-marathe I had a quick look, didn't dive into the PR but I guess it uses Impress Tile Layer and file based view implementation. We can today, say:

**a new view => a-new-view-layout.ts file.**

If we have time for this feature, I would like to see another view layout implementation for this feature if possible.

`ViewLayoutCompareChanges.ts`, `ViewLayoutMultiPage.ts` files and `DocumentBase.ts`  file may help getting started for a new view implementation if you wish (`ViewLayoutImpressMultiPage.ts`)?

Please let me know if I can help with anything. Thank you.

**Comment by mohit-marathe** [comment]
> [@mohit-marathe](https://github.com/mohit-marathe) I had a quick look, didn't dive into the PR but I guess it uses Impress Tile Layer and file based view implementation. We can today, say:
> 
> **a new view => a-new-view-layout.ts file.**
> 
> If we have time for this feature, I would like to see another view layout implementation for this feature if possible.
> 
> `ViewLayoutCompareChanges.ts`, `ViewLayoutMultiPage.ts` files and `DocumentBase.ts` file may help getting started for a new view implementation if you wish (`ViewLayoutImpressMultiPage.ts`)?
> 
> Please let me know if I can help with anything. Thank you.

Thank you @gokaysatir for the pointers! I will read more about this, and see how will this layout work in `Editing mode`.

**Comment by mmeeks** [comment]
Nice to have the PDF thing there; I guess we would want page-up/down to be page-informed and jump slides nicely if that doesn't work currently. Nice to see it's a one-liner ;-)

**Comment by mohit-marathe** [comment]
This feature has already been merged. In Desktop, it is available in `Read-Only` mode. While in mobile, it's available in both `Read-Only`, and `Viewing mode`.

**Comment by mohit-marathe** [comment]
> Nice to have the PDF thing there; I guess we would want page-up/down to be page-informed and jump slides nicely if that doesn't work currently. Nice to see it's a one-liner ;-)

Currently, I'm creating a view layout for fileBasedView - https://gerrit.collaboraoffice.com/c/online/+/2952 . so i'll address that there.
